src/robo_snap.py

```diff
@@ -177,18 +178,49 @@ def start_snap(self) -> None:
 
         try:
             max_score = -1  # score初期値
-            for _ in range(len(self.img_list)):
+            start_time = 0.0  # 次の画像の受信開始前のタイムスタンプ
+            # 次のスナップまでの各区間におけるタイムアウト時間リスト
+            timeouts = [
+                Timeouts.TIMEOUT_FROM_FigA_1_TO_FigA_2,  # i = 1 の時、timeouts[0]
+                Timeouts.TIMEOUT_FROM_FigA_2_TO_FigB,   # i = 2 の時、timeouts[1]
+                Timeouts.TIMEOUT_FROM_FigB_TO_FigA_3,   # i = 3 の時、timeouts[2]
+                Timeouts.TIMEOUT_FROM_FigA_3_TO_FigA_4  # i = 4 の時、timeouts[3]
+            ]
+            timeout_flag = False  # タイムアウトしたかどうかを判定するフラグ
+
+            for i in range(len(self.img_list)):
 
                 # 走行体から画像を取得
                 while True:  # 画像が見つかるまでループ
                     # 画像の受信試み
                     img_name, img_path = self.scp_fig_image()
 
+                    # 画像を受信したらループを抜ける
                     if img_name is not None:
+                        # FigA_4でなければ、計測を開始する.
+                        if img_name != "FigA_4.png":
+                            start_time = time.time()
                         break
 
+                    # 画像を受信しておらず、計測が開始されている時の処理
+                    elif start_time != 0.0:

```

Comment:
```suggestion
                    elif i != 0:
```
最初のループって意味ならiを使うべきだと思う。